src/OrchardCore.Modules/OrchardCore.Media/Services/MediaCacheBackgroundTask.cs

```diff
@@ -0,0 +1,153 @@
+        {
+            if (!baseDir.Exists)
+            {
+                return;
+            }
+
+            try
+            {
+                var dirs = baseDir.EnumerateDirectories().ToArray();
+                foreach (var dir in dirs)
+                {
+                    try
+                    {
+                        await RecursiveDeleteAsync(dir, true, cancellationToken);
+                    }
+                    catch (Exception ex)
+                    {
+                        _logger.LogError(ex, "Error deleting dir {DirName}", dir.Name);
+                    }
+                }
+            }
+            catch (Exception ee)
+            {
+                _logger.LogError(ee, "Error enumerating dirs {DirName}", baseDir.Name);
+            }
+
+            var files = baseDir.GetFiles();
+            foreach (var file in files)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                try
+                {
+                    file.IsReadOnly = false;
+
+                    if (IsFileLocked(file))
+                    {
+                        var i = 0;
+                        while (file.Exists && IsFileLocked(file) && i <= Repeats)
+                        {
+                            await Task.Delay(RepeatTime, cancellationToken);
+                            file.Delete();
+                            i++;
+                        }
+                    }
+                    else
+                    {
+                        file.Delete();
+                    }
+                }
+                catch (Exception e)
+                {
+                    _logger.LogError(e, "Error deleting cache file {FilePath}", file.Name);
+                }
+            }
+
+            if (deleteBaseDir)
+            {
+                try
+                {
+                    var i = 0;
+                    while (baseDir.Exists && baseDir.GetFiles().Length == 0 && i <= Repeats)
+                    {
+                        baseDir.Delete();
+                        await Task.Delay(RepeatTime, cancellationToken);
+                        i++;
+                    }
+                }
+                catch (Exception e)
+                {
+                    _logger.LogError(e, "Error deleting cache folder {DirectoryPath}", baseDir.Name);
+                }
+            }
+        }
+
+        private static bool IsFileLocked(FileInfo file)
+        {
+            FileStream stream = null;
+
+            try
+            {
+                stream = file.Open(FileMode.Open, FileAccess.ReadWrite, FileShare.None);
+            }
+            catch (IOException)
+            {
+                //the file is unavailable because it is:
+                //still being written to
+                //or being processed by another thread
+                //or does not exist (has already been processed)
```

Comment:
```suggestion
                // The file is unavailable because it is:
                // still being written to,
                // or being processed by another thread,
                // or does not exist (has already been processed).
```

Reply:
You don't need to add a "," when using "or" in an enumeration like in french.

Reply:
Okay, so just the missing spaces after `//` and the Upercase at the beginning, and the dot at the end

TBH i don't care, just because i'm now used to do it, so anyway as you want ;)